src/body.rs

```diff
@@ -31,6 +42,19 @@ impl Body {
     /// advance, but where we don't want to load the data in memory.  This
     /// is useful if we need to ensure `Content-Length` is passed with the
     /// request.
+    ///
+    /// ```rust
+    /// # use reqwest::Body;
+    /// # fn run() -> Result<(), Box<std::error::Error>> {
+    /// // &[u8] implements std::io::Read, and the source `s` has a
+    /// // 'static lifetime and a known number of bytes.
```

Comment:
I think the example probably should address the point in the lines above, about how this constructor is useful if "we don't want to load the data in memory". If someone has a a bundle of bytes, it's actually better in to use `Body::from()`.

This was originally added to allow someone to include the length with a `File` or similar.

```rust
let file = File::open("foo.txt")?;
let len = file.metadata()?.len();
let body = Body::sized(file, len);
```